tests/sklearn/test_sklearn_autolog.py

```diff
@@ -0,0 +1,392 @@
+    np.testing.assert_array_equal(loaded_model.predict(Xy[0]), model.predict(Xy[0]))
+
+
+def test_meta_estimator():
+    mlflow.sklearn.autolog()
+
+    estimators = [
+        ("std_scaler", sklearn.preprocessing.StandardScaler()),
+        ("svc", sklearn.svm.SVC()),
+    ]
+    model = sklearn.pipeline.Pipeline(estimators)
+    Xy = get_iris()
+
+    with mlflow.start_run() as run:
+        model.fit(*Xy)
+
+    run_id = run._info.run_id
+    params, metrics, tags, artifacts = get_run_data(run_id)
+    assert params == stringify_dict_values(model.get_params(deep=True))
+    assert metrics == {TRAINING_SCORE: model.score(*Xy)}
+    assert tags == {
+        ESTIMATOR_NAME: model.__class__.__name__,
+        ESTIMATOR_CLASS: model.__class__.__module__ + "." + model.__class__.__name__,
+    }
+    assert "model" in artifacts
+
+    loaded_model = load_model_by_run_id(run_id)
+    np.testing.assert_array_equal(loaded_model.predict(Xy[0]), model.predict(Xy[0]))
+
+
+def test_get_params_returns_dict_that_has_more_keys_than_max_params_tags_per_batch():
+    mlflow.sklearn.autolog()
+
+    large_params = {str(i): str(i) for i in range(MAX_PARAMS_TAGS_PER_BATCH + 1)}
+    Xy = get_iris()
+
+    with mock.patch("sklearn.cluster.KMeans.get_params", return_value=large_params):
+        with mlflow.start_run() as run:
+            model = sklearn.cluster.KMeans()
+            model.fit(*Xy)
+
+    run_id = run._info.run_id
+    params, metrics, tags, artifacts = get_run_data(run._info.run_id)
+    assert params == large_params
+    assert metrics == {TRAINING_SCORE: model.score(*Xy)}
+    assert tags == {
+        ESTIMATOR_NAME: model.__class__.__name__,
+        ESTIMATOR_CLASS: model.__class__.__module__ + "." + model.__class__.__name__,
+    }
+    assert "model" in artifacts
+
+    loaded_model = load_model_by_run_id(run_id)
+    np.testing.assert_array_equal(loaded_model.predict(Xy[0]), model.predict(Xy[0]))
+
+
+def test_get_params_returns_dict_with_value_longer_than_max_param_val_length():
+    mlflow.sklearn.autolog()
+
+    long_params = {"name": "a" * (MAX_PARAM_VAL_LENGTH + 1)}
+    Xy = get_iris()
+
+    with mock.patch("sklearn.cluster.KMeans.get_params", return_value=long_params):
+        with mlflow.start_run() as run:
+            model = sklearn.cluster.KMeans()
+            model.fit(*Xy)
+
+    run_id = run._info.run_id
+    params, metrics, tags, artifacts = get_run_data(run._info.run_id)
+    assert params == {"name": "a" * MAX_PARAM_VAL_LENGTH}
+    assert metrics == {TRAINING_SCORE: model.score(*Xy)}
+    assert tags == {
+        ESTIMATOR_NAME: model.__class__.__name__,
+        ESTIMATOR_CLASS: model.__class__.__module__ + "." + model.__class__.__name__,
+    }
+    assert "model" in artifacts
+
+    loaded_model = load_model_by_run_id(run_id)
+    np.testing.assert_array_equal(loaded_model.predict(Xy[0]), model.predict(Xy[0]))
+
+
+def test_call_fit_with_arguments_score_does_not_accept():
```

Comment:
Awesome! Can we also add a small test for the case where `X` and `y` are specified as kwargs?